contrib/pg_tde/src/access/pg_tde_xlog_encrypt.c

```diff
@@ -179,13 +179,14 @@ TDEXLogSmgrInit(void)
 								  ((key->rel_type & TDE_KEY_TYPE_WAL_ENCRYPTED && !EncryptXLog) ||
 								   (key->rel_type & TDE_KEY_TYPE_WAL_UNENCRYPTED && EncryptXLog))))
 	{
-		pg_tde_create_wal_key(
-							  &EncryptionKey, &GLOBAL_SPACE_RLOCATOR(XLOG_TDE_OID),
+		pg_tde_create_wal_key(&EncryptionKey, &GLOBAL_SPACE_RLOCATOR(XLOG_TDE_OID),
 							  (EncryptXLog ? TDE_KEY_TYPE_WAL_ENCRYPTED : TDE_KEY_TYPE_WAL_UNENCRYPTED));
+		EncryptionContext = NULL;
 	}
 	else if (key)
 	{
 		EncryptionKey = *key;
+		EncryptionState = NULL;
```

Comment:
`EncryptionState` is a pointer to shared memory allocated in `TDEXLogShmemInit()`. It should be set to NULL here

Reply:
Wow, that was a bad typo. Suprisede no tests broke.